### ISSUE TITLE
Crash when trying to open epub file

### ISSUE THREAD

**Issue by FattyNess** [dstark5/Openlib#22]
![Screenshot_20230904-205140](https://github.com/dstark5/Openlib/assets/37911092/22617df0-0707-45f7-83ae-edc3a3e673a6)

App has access to files, but crushes when trying to open epub file.

Android 11, OxygenOS.

**Comment by dstark5** [comment]
Did you tried to uninstall and reinstall the app

**Comment by FattyNess** [comment]
Just have done that. Doesn't help. Reinstalled it from releases and from f-droid, nothing changes.

**Comment by dstark5** [comment]
after reinstalling the app and opening epub will ask you for file permission did you granted it

**Comment by ChandanShakya** [comment]
On my Redmi Phone, it doesnt show the error but on my samsung phone it shows that error, and doesnt allow reading epub files, i have given all permissions.
Redmi Note 10S and Samsung M31

**Comment by dstark5** [comment]
did you grant storage access while opening epub

**Comment by ChandanShakya** [comment]
> did you grant storage access while opening epub

Yessss i did.

**Comment by dstark5** [comment]
> > did you grant storage access while opening epub
> 
> Yessss i did.

Does it worked
If not what error did you get

**Comment by ChandanShakya** [comment]
> > > did you grant storage access while opening epub
> > 
> > 
> > Yessss i did.
> 
> Does it worked If not what error did you get

Now it works, i grant the files access and allowed it to read all files.

Thus, Works on My Redmi Note 10S and Samsung Galaxy M31

**Comment by dstark5** [comment]
> > after reinstalling the app and opening epub will ask you for file permission did you granted it
> 
> I've done that, actually.
> 
> > If it still crashes , Kindly turn on Open Epub With External reader in settings page and use external reader .I will fix it ASAP in upcoming release
> 
> For now, I am using Librera to open epub files. Waiting for the fix, because I don't like librera's design:)

will sure fix it ASAP
Thank you

**Comment by dstark5** [comment]
> > > > did you grant storage access while opening epub
> > > 
> > > 
> > > Yessss i did.
> > 
> > 
> > Does it worked If not what error did you get
> 
> Now it works, i grant the files access and allowed it to read all files.
> 
> Thus, Works on My Redmi Note 10S and Samsung Galaxy M31 Ps: i installed latest 1.0.2 version for v8a

great 😊

**Comment by powerfulparadox** [comment]
My Galaxy Tab S6 just developed this issue with the latest version. It worked fine before. Reinstalling and re-granting the permission did not fix the issue.

My Galaxy A52 5G works fine with the latest version.

**Comment by dstark5** [comment]
What android version does your tab runs on

**Comment by powerfulparadox** [comment]
> What android version does your tab runs on

12

**Comment by dstark5** [comment]
did the app crashed or shows 'cannot open epub need storage permission' error

**Comment by powerfulparadox** [comment]
It crashes to launcher and shows the message "cannot open epub it needs storage access!"

**Comment by justpankajdotcom** [comment]
HI @dstark5 , does it have any solution yet? I am also facing the same issue while opening EPUB documents in my moto g32, running android 12.

**Comment by dstark5** [comment]
Yeah will sure fix the issue you mentioned in next update as the version 1.0.3 is a solution for anna's archive captcha

**Comment by justpankajdotcom** [comment]
Thank you, looking forward to the stable version, @dstark5 . App has a very beautiful interface, kudos !
